sympy/printing/codeprinter.py

```diff
@@ -14,6 +16,32 @@
 from sympy.codegen.ast import Assignment
 
 
+def prints_statement(method):
+    """ Decorator for methods printing statements (as opposed to expressions) """
+
+    @wraps(method)
+    def wrapper(self, expr):
+        if self._settings.get('enable_statements', True):
+            return method(self, expr)
+        else:
+            raise ValueError("The code-printer setting 'enable_statements' is False,\n"
+                             "but %s generates a statement in the chosen language." %
+                             type(expr))
```

Comment:
Maybe both are misleading because neither expressions or statements are a subset of the other in general. Even in C, for instance, you can't really say an expression is a statement because you need a semicolon to make a statement. And that matters because if you have `C_CODE_TEMPLATE.format(expr=ccode(expr))` then it matters if `expr` should be printed as an expression or statement simply by the question of whether ccode should print a semicolon or not.